maven-core/src/main/java/org/apache/maven/MissingProfileException.java

```diff
@@ -0,0 +1,32 @@
+package org.apache.maven;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * Signals that the user referenced a Maven profile that could not be located in either the project or the settings.
+ */
+public class MissingProfileException
+        extends Exception
```

Comment:
Referencing a profile in your Maven invocation that does not exist is an error made by the user. The system (Maven) is not in an illegal state. So it could be `IllegalArgumentException` if you insist.

Reply:
Agreed, I take the IAE.